src/carp.cc

```diff
@@ -106,14 +93,10 @@ carpInit(void)
 
         if (floor(p->carp.load_factor * 1000.0) == 0.0)
             p->carp.load_factor = 0.0;
-
-        /* add it to our list of peers */
-        *P = cbdataReference(p);
-        ++P;
     }
 
     /* Sort our list on weight */
-    qsort(carp_peers, n_carp_peers, sizeof(*carp_peers), peerSortWeight);
+    qsort(rawCarpPeers.data(), rawCarpPeers.size(), sizeof(RawCachePeers::value_type), peerSortWeight);
```

Comment:
We might have used decltype() here instead.

Reply:
We could write something like  `sizeof(decltype(rawCarpPeers)::value_type)` or `sizeof(decltype(rawCarpPeers.back()))`: it would protect from changing rawCarpPeers type and forgetting to change the explicit type in sizeof().

Reply:
Good point! Let's try the first variant -- ` sizeof(decltype(rawCarpPeers)::value_type)`. I think it is more direct/obvious. However, if that does not work well, I am not against the second variant, although I would probably use front() instead of back() there.

Reply:
Done.